src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1890,3 +1928,44 @@ fn is_primitive(path_str: &str, ns: Namespace) -> Option<(&'static str, Res)> {
 fn primitive_impl(cx: &DocContext<'_>, path_str: &str) -> Option<&'static SmallVec<[DefId; 4]>> {
     Some(PrimitiveType::from_symbol(Symbol::intern(path_str))?.impls(cx.tcx))
 }
+
+fn strip_generic_parameters_from_path(
+    path_str: &str,
+) -> Result<String, ResolutionFailure<'static>> {
+    let mut stripped_segments = vec![];
+
+    for segment in path_str.split("::") {
+        let stripped_segment = strip_generic_parameters_from_path_segment(segment)?;
```

Comment:
Since this strips _every_ path segment, it allows things that aren't valid in the type system, like `Vec::new<T>()`. I don't think that's a big deal (it's more flexible, not less), but it would be nice to add some test cases for it to make sure it doesn't change by accident.

Reply:
What kind of tests do you mean?

Reply:
Like `[Vec::new::<T>()]` and `[Vec::new<T>()]`. Both of those aren't valid according to the type system but will be allowed by this change.

Reply:
By the way, I think this change will also allow things like `Vec<<T>>` and `Vec<>`, but I'm not sure it's worth checking for that since it's kind of obvious to the user.

Reply:
I'd really prefer to test things even if they're not explicitly supported. I ran into this issue in https://github.com/rust-lang/rust/pull/73566 - the things `everybody_loops` allowed were never tested, so it was very unclear which changes would and wouldn't break things (it caused a P-critical: https://github.com/rust-lang/rust/issues/75100). It's less of an issue here because `broken_intra_doc_links` is only a warning, but I'd still prefer to have a larger test suite.

Reply:
Okay, you mean to add tests not because we want to support something but just so we know if what we _implicitly_ support changes? That sounds good to me.

Reply:
Exactly. Maybe add a comment too like 'this is the current behavior, but not guaranteed to be stable'.

Reply:
In the test you mean?

Reply:
Yes, in the test.